manifests/virtual_machine.pp

```diff
@@ -93,7 +93,12 @@
     ensure => 'directory',
   }
 
-  unless $facts['vm_guests'].member($title) {
+  if $facts['vm_guests'].member($title) {
+    exec { "${prefix}::autostart":
+      creates => "${autostart_path}/${title}.xml",
+      command => "/usr/bin/virsh autostart ${title}",
+    }
+  } else {
```

Comment:
In case this is tricky to parse, here's my attempt at translating to English. It was:

1. If the VM doesn't exist, put an `exec` that creates the VM in the host catalog.
2. If the VM doesn't exist, put an `exec` in the catalog that ensures the VM is autostart, but `require` the install `exec`.

This was a problem because often, creating a VM times out and succeeds outside of the puppet run. So the autostart `exec` never ran on the initial run, and on subsequent runs it was in the wrong condition branch, since the VM exists now.

Now it's

1. If the VM doesn't exist, put an `exec` that creates the VM in the host catalog.
2. If the VM *does* exist, put an `exec` in the catalog that ensures the VM is autostart (with no `require`).

Reply:
And we can't run this outside of the condition statement (rather than moving from the `unless` to `if`) because we'd need a `require` in one case and not the other?

This is a bit odd, but I think it's fine. I'd suggest adding a comment like:
```
# ensure autostart settings on vm, after creation on a previous puppet run
if
...
# create vm, only if it doesn't exist
else
```

Reply:
If this runs before the VM is created, I'm not sure what its behavior would look like. I doubt it would cause lasting damage, but it also wouldn't do anything valid.